### ISSUE TITLE
[Feature request] Support multiple elasticsearch nodes

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#73449]
### Discussed in https://github.com/grafana/grafana/discussions/58399

<div type='discussions-op-text'>

<sup>Originally posted by **dploeger** April 19, 2017</sup>
Grafana version: 4.1.1
Datasource: Elasticsearch 2.3.5
OS: macOS 10.12.4

The elasticsearch data source currently only seems to support a single ES-node. Usually, elasticsearch comes as a cluster with multiple nodes. It would be great, if one could specify multiple nodes, either in a master/fallback a real round-robin way.

The other possibility would be to create a high available load balancer in front of them, but it would be much nicer and easier, if the application supports it out of the box.</div>

**Comment by adamzanyline** [comment]
Hello!
As Elasticsearch has this functionality already implemented per-se (as "[coordinating node](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html#coordinating-node)") I believe re-implementing round-robin or so from within Grafana would be redundant.

**Comment by morrone** [comment]
> Hello! As Elasticsearch has this functionality already implemented per-se (as "[coordinating node](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html#coordinating-node)") I believe re-implementing round-robin or so from within Grafana would be redundant.

If you read that closely, you'll see that _all_ elasticsearch nodes are coordinating nodes, implicitly. That link is saying that you _may_ set up additional nodes where you disable all other services, leaving only the coordinating node capability remaining enabled. Not all clusters will have that. And even if a cluster has coordinating-only nodes, that doesn't necessarily address all of the use cases for listing multiple nodes in grafana's configuration of an elasticsearch data source.

It does appear to me that in the elastic stack realm it is normal for elasticsearch clients to take a list of elasticsearch servers rather than just one, and adding this capability to grafana would bring it in line with normal use practices.

**Comment by maosmurf** [comment]
You are right as in that this is one use-case, and indeed still supported using the [elasticsearch clients](https://www.elastic.co/guide/en/elasticsearch/client/index.html)  (e.g. in TS, allowing `string[]` as node parameter in [ClientOptions](https://github.com/elastic/elasticsearch-js/blob/main/src/client.ts#L83), equally for other languages).

**Comment by frittentheke** [comment]
As it was already discussed in https://github.com/grafana/grafana/discussions/58399, Grafana does not use a client lib to connect to either ElasticSearch or OpenSearch, but their self-crafted HTTP / REST client (see https://github.com/grafana/grafana/blob/f46c07aba7b6faccd2ecafc83051d1410cacc867/pkg/tsdb/elasticsearch/elasticsearch.go.

The client libs actually support multiple nodes out of the box (and likely other improvements in efficiently communicating with Elastic- or OpenSearch:

* ElasticSearch 
  * https://www.elastic.co/guide/en/elasticsearch/client/go-api/current/connecting.html#connecting-multiple-nodes
* OpenSearch
  * https://opensearch.org/docs/latest/clients/go/
  * https://pkg.go.dev/github.com/opensearch-project/opensearch-go/v4#Client.DiscoverNodes
